R/provider-google.R

```diff
@@ -687,20 +709,29 @@ models_google_gemini <- function(
 
   json <- resp_body_json(resp)
 
+  if (isFALSE(api_key)) { # if Vertex
+    name <- map_chr(json$publisherModels, "[[", "name")
+    name <- gsub("^publishers/google/models/", "", name)
+    # this is the closest to "generateContent" in "supportedGenerationMethods" for Gemini
+    # https://cloud.google.com/vertex-ai/docs/reference/rest/v1beta1/publishers.models
+    can_generate <- json$publisherModels |> map_lgl(\(x) "openGenerationAiStudio" %in% names(x$supportedActions))
+  } else {
```

Comment:
Nit: I think the indentation is off here now. Can you run this through `air` (if you have it).

Reply:
Sure, will do.  I was using tidyverse styler.